src/NHibernate/Loader/Criteria/CriteriaQueryTranslator.cs

```diff
@@ -863,6 +863,24 @@ public string GetEntityName(ICriteria subcriteria, string propertyName)
 			return GetEntityName(subcriteria);
 		}
 
+		internal IDictionary<string, string> GetCriteriaSQLAliasMap()
+		{
+			var result = outerQueryTranslator != null
+				? new Dictionary<string, string>(outerQueryTranslator.GetCriteriaSQLAliasMap())
```

Comment:
Creating a new dictionary because `outerQueryTranslator.GetCriteriaSQLAliasMap()` could be an `EmptyDictionary` which is read-only.